bin/logstash.lib.sh

```diff
@@ -75,12 +75,28 @@ parse_jvm_options() {
 setup_java() {
   # set the path to java into JAVACMD which will be picked up by JRuby to launch itself
   if [ -z "$JAVACMD" ]; then
-    if [ -x "$JAVA_HOME/bin/java" ]; then
-      JAVACMD="$JAVA_HOME/bin/java"
+    JAVACMD_TEST=`command -v java`
```

Comment:
that if it has been moved below https://github.com/elastic/logstash/pull/11845/files/6d36538c2bd2f94634cf8137723bb16e5784aa2d#diff-153c0379c26855674ab54f42c924c7efR93.

The flow should be if no JAVA_HOME and no Java are present in the system use the bundled (determining the correct path in case of Darwin) else detect if it's present JAVA_HOME or the java command and use that

Reply:
Chatting with @yaauie, he came out with an idea. We could use a different priority, so first check `JAVA_HOME`, if not exists use the bundled JDK (if present) then fallback to `command -v java` to check if any Java is installed, else raise an error message.
In case of bundled JDK if the user is defining a JAVA_HOME we could log a message that he is using a Logstash bundled with JDK and use a customized JAVA_HOME, so warning him that probably he could use a Logstash distribution without  bundled JDK. WDYT?

Reply:
Yeah, that makes sense to me. It makes the choice of using the bundled Java or a custom Java relatively straightforward and logging the fact that the running Java is sourced via `JAVA_HOME` helps surface users potentially inadvertently using a custom Java.

Reply:
done